src/main/java/org/jabref/logic/util/io/FileFinder.java

```diff
@@ -17,10 +19,9 @@
      * @param directories The root directories to search.
      * @param extensions  The extensions that are acceptable.
      */
-    Map<BibEntry, List<Path>> findAssociatedFiles(List<BibEntry> entries, List<Path> directories, List<String> extensions);
+    Multimap<BibEntry, Path> findAssociatedFiles(List<BibEntry> entries, List<Path> directories, List<String> extensions);
 
-    default List<Path> findAssociatedFiles(BibEntry entry, List<Path> directories, List<String> extensions) {
-        Map<BibEntry, List<Path>> associatedFiles = findAssociatedFiles(Collections.singletonList(entry), directories, extensions);
-        return associatedFiles.getOrDefault(entry, Collections.emptyList());
+    default Collection<Path> findAssociatedFiles(BibEntry entry, List<Path> directories, List<String> extensions) {
```

Comment:
The multimap only returns a collection because it actually is not implemented as a list (the sorting is random). But now we don't need multimaps anymore so we can go back to lists!